test/test_util.py

```diff
@@ -48,6 +48,7 @@
 # numbers used for timeouts
```

Comment:
As I mentioned above, if you look inside that dictionary there is two duplicated keys with different values within that dictionary so for the entire time the test has been around two of the values were not being tested and silently discarded.

```python
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> a = {'a': 1, 'a': 2}
>>> a
{'a': 2}
```

By changing this to a list of tuples it preserves both test cases and changing to a list of tuples necessitates removing the `.items()` the bottom loop where these values are used.

I'm open to moving this to it's own PR but as it stands currently this test is dropping some of it's test cases silently, something I figured we'd want to avoid.